File: src/main/java/it/contrader/services/AbstractService.java
package it.contrader.services;

import java.util.ArrayList;
import java.util.List;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Service;

import it.contrader.converter.Converter;

@Service
public abstract class AbstractService<Entity,DTO> implements MyService<Entity,DTO> {
	
	protected CrudRepository<Entity,Long> crudRepository;
	protected Converter<Entity,DTO> converter;
	
	public AbstractService() {
	}

	@Override
	public Entity insert(DTO dto) {
		return crudRepository.save(converter.toEntity(dto));
	}

	@Override
	public List<DTO> getAll() {
		Iterable<Entity> iter = crudRepository.findAll();
		List<DTO> list = new ArrayList<>();
		for (Entity item : iter) {
	        list.add(converter.toDTO(item));
	    }
		return list;
	}

	@Override
	public DTO read(long id) {
		return converter.toDTO(crudRepository.findById(id).get());
	}

	@Override
	public Entity update(DTO dto) {
		return crudRepository.save(converter.toEntity(dto));
	}

	@Override
	public void delete(DTO dto) {
		crudRepository.delete(converter.toEntity(dto));
	}
}

File: src/app/services/abstractservice.ts
import { Service } from './service';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

export abstract class AbstractService<DTO> implements Service<DTO> {

    type: string;

    constructor(protected http: HttpClient) {

    }

      getAll(): Observable<DTO[]> {
      return this.http.get<DTO[]>('http://localhost:8080/' + this.type + '/getall');
  }

  read(id: number): Observable<DTO> {
      return this.http.get<DTO>('http://localhost:8080/' + this.type + '/read?id=' + id);
  }

  delete(id: number): Observable<any> {
      return this.http.delete('http://localhost:8080/' + this.type + '/delete?id=' + id);
  }

  insert(apartment: DTO): Observable<any> {
      return this.http.post('http://localhost:8080/' + this.type + '/insert', apartment);
  }

  update(apartment: DTO): Observable<any> {
      return this.http.put('http://localhost:8080/' + this.type + '/update', apartment);
  }
}
